mammoth/bin/train.py

```diff
@@ -59,45 +59,45 @@
 
 # TODO: reimplement save_transformed_sample
 
-def _init_train(opts):
-    """Common initilization stuff for all training process."""
-    ArgumentParser.validate_prepare_opts(opts)
-
-    if opts.train_from:
-        # Load checkpoint if we resume from a previous training.
-        checkpoint = load_checkpoint(ckpt_path=opts.train_from)
-        # fields = load_fields(opts.save_data, checkpoint)
-        transforms_cls = get_transforms_cls(opts._all_transform)
-        if (
-            hasattr(checkpoint["opts"], '_all_transform')
-            and len(opts._all_transform.symmetric_difference(checkpoint["opts"]._all_transform)) != 0
-        ):
-            _msg = "configured transforms is different from checkpoint:"
-            new_transf = opts._all_transform.difference(checkpoint["opts"]._all_transform)
-            old_transf = checkpoint["opts"]._all_transform.difference(opts._all_transform)
-            if len(new_transf) != 0:
-                _msg += f" +{new_transf}"
-            if len(old_transf) != 0:
-                _msg += f" -{old_transf}."
-            logger.warning(_msg)
-        # if opts.update_vocab:
-        #    logger.info("Updating checkpoint vocabulary with new vocabulary")
-            # fields, transforms_cls = prepare_fields_transforms(opts)
-    else:
-        checkpoint = None
-        # fields, transforms_cls = prepare_fields_transforms(opts)
-
-    # Report src and tgt vocab sizes
-    # for side in ['src', 'tgt']:
-    #     f = fields[side]
-    #     try:
-    #         f_iter = iter(f)
-    #     except TypeError:
-    #         f_iter = [(side, f)]
-    #     for sn, sf in f_iter:
-    #         if sf.use_vocab:
-    #             logger.info(' * %s vocab size = %d' % (sn, len(sf.vocab)))
-    return checkpoint, None, transforms_cls
+# def _init_train(opts):
```

Comment:
any good reason to keep as a comm?

Reply:
Nah, I'll remove it.

Reply:
There was a "TODO: reimplement save_transformed_sample".
I guess the idea of that feature is checking what a transform pipeline actually does to the data, if it is acting funky.
Doesn't work atm, so I guess I'll just remove it.

Reply:
The opt flag was actually called `--dump_samples`